client/state/data-layer/wpcom/concierge/schedules/appointments/book/index.js

```diff
@@ -56,17 +57,25 @@ export const onError = ( { type }, error ) => {
 			? 'calypso_concierge_appointment_rescheduling_error'
 			: 'calypso_concierge_appointment_booking_error';
 
-	const errorMessage =
-		CONCIERGE_ERROR_NO_AVAILABLE_STAFF === error.code
-			? translate( 'This session is no longer available. Please select a different time.' )
-			: translate( 'We could not book your appointment. Please try again later.' );
+	const errorMessage = code => {
```

Comment:
If we're going to turn the message generator into a pure function like this, we can probably move it outside this function scope so it's not re-created every time `onError` is called.